curl/src/client.cc

```diff
@@ -255,12 +282,22 @@ namespace leatherman { namespace curl {
         // Perform the request
         auto result = curl_easy_perform(_handle);
         if (result == CURLE_WRITE_ERROR) {
-          throw http_file_operation_exception(req, file_path, make_file_err_msg(_("failed to write to the temporary file during download")));
+            throw http_file_operation_exception(req, file_path, make_file_err_msg(_("failed to write to the temporary file during download")));
         } else if (result != CURLE_OK) {
-          throw http_file_download_exception(req, file_path, _("File download server side error: {1}", errbuf));
+            throw http_file_download_exception(req, file_path, _("File download server side error: {1}", errbuf));
+        }
+
+        // Check the status code. If 400+, fill in the response
+        LOG_DEBUG("request completed (status {1}).", _res.status_code());
+        if (_res.status_code() > 400 && res) {
```

Comment:
How does a caller know when there's something to read in response? I was imaging `res` would only be enabled if status code was `>= 400` (also, needs to be `>= 400`, `not > 400`). It looks like we'll always initialize it if provided, so we'll query the status code?